File: examples/spring-io-2026/java/src/main/java/org/example/koog/spring/java/structs/UserAccountInfo.java
package org.example.koog.spring.java.structs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;

public record UserAccountInfo(
        @JsonProperty("username") String username,
        @JsonProperty("email") String email,
        @JsonProperty("phoneNumber") String phoneNumber,
        @JsonProperty("address") String address,
        @JsonProperty("accountNumber") String accountNumber,
        @JsonProperty("accountType") String accountType,
        @JsonProperty("balance") Integer balance,
        @JsonProperty("createdAt") Instant createdAt,
        @JsonProperty("lastTransactionTime") Instant lastTransactionTime
) {
    @JsonCreator
    public UserAccountInfo {}
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/structs/UserAccountInfo.kt
package org.example.koog.spring.kotlin.structs

import com.fasterxml.jackson.annotation.JsonCreator
import kotlin.time.Instant
import kotlinx.serialization.Serializable

@Serializable
data class UserAccountInfo @JsonCreator constructor(
    val username: String,
    val email: String,
    val phoneNumber: String,
    val address: String,
    val accountNumber: String,
    val accountType: String,
    val balance: Int,
    val createdAt: Instant,
    val lastTransactionTime: Instant
)
